cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -281,6 +356,24 @@ func buildPartitionList(ptable *sfdiskPartitionTable, pv *gadget.LaidOutVolume)
 	return buf, toBeCreated
 }
 
+// listCreatedPartitions returns a list of partitions created during the
+// install process.
+// TODO:UC20: also provide a mechanism for MBR (RPi)
+func listCreatedPartitions(dl *DeviceLayout) []string {
+	created := make([]string, 0, len(dl.partitionTable.Partitions))
+	for _, p := range dl.partitionTable.Partitions {
+		for _, a := range strings.Fields(p.Attrs) {
+			if idx := strings.Index(a, "GUID:"); idx >= 0 {
+				attrs := strings.Split(a[idx+5:], ",")
+				if strutil.ListContains(attrs, createdPartitionAttr) && strutil.ListContains(createdPartitionGUID, strings.ToUpper(p.Type)) {
+					created = append(created, p.Node)
+				}
+			}
+		}
+	}
```

Comment:
Nitpick, this could be simplified a bit:
```suggestion
	for _, p := range dl.partitionTable.Partitions {
		if !strutil.ListContains(createdPartitionGUID, strings.ToUpper(p.Type)) {
			continue
		}
		idx := strings.Index(p.Attrs, "GUID:")
		if idx == -1 {
			continue
		}
		guidAttrsField := strings.Fields(p.Attrs[idx+5:])[0]
		attrs := strings.Split(guidAttrsField, ",")
		if strutil.ListContains(attrs, createdPartitionAttr) {
			created = append(created, p.Node)
		}
	}

```

Reply:
Now Github allows me to comment here: the GUID: test must be in each result field instead of always at the start of the attributes string but I got the idea.